packages/amplify_datastore_plugin_interface/lib/src/types/models/uuid.dart

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import 'package:uuid/uuid.dart';
+
+class UUID {
+  static final _internal = Uuid();
+
+  static String getUUID() {
+    return _internal.v4();
+  }
+}
```

Comment:
Do we need this utility? Why not just use `Uuid().v4()` in the code?

Reply:
The Amplify data modeling suite does include an entity called an `ID`. To strongly model that type, it _could_ make sense to have a thing called an `ID`. (After all, we have things such as `Temporal.DateTime`, etc., that come from the modeling SDL.)

Today the Android code base just uses `UUID.randomUUID().toString()` in a variety of places. There's nothing really wrong with that, but it would a hair cleaner to encapsulate the UUID details into the Amplify modeling type (an "`ID`", whatever that is.)

Reply:
I think that would expose the implementation details too much especially if we wanted to change the way in which we generate a UUID.  I prefer now to call directly third party code in this way.